Automattic/simplenote-ios
# Added automated UI tests to the workspace

### Fix
During my trial I automated 21 UI test, and kept them in my fork. Now created a branch here. Having these tests in develop will help running automated tests against next betas.

I understand this is a huge one. I wrote this for two weeks only in my fork. But committing this in parts in current state will make tests execution impossible.

### Test
Works best on iPhone 12. Certain tests will fail on previous versions, because of different ways to locate controls. I'm planning to add support of multiple versions gradually.

1. Make sure that 'Use the Same Keyboard Language as macOS' and 'Connect Hardware Keyboard' are disabled in Simulator properties ('I/O' → 'Keyboard'). 
2. All tests are located in SimplenoteUITests.swift
3. Run them in similar way as existing 'SimplenoteTests' or 'SimplenoteScreenshots'

### Review
Besides new files with tests themselves, please check if I did not break anything 'unrelated':
- .bundle/config
- Simplenote.xcodeproj/project.pbxproj
- Simplenote.xcodeproj/xcshareddata/xcschemes/Simplenote.xcscheme
- Simplenote.xcodeproj/xcshareddata/xcschemes/SimplenoteScreenshots.xcscheme (have no idea why this was affected - maybe because I created one more 'UI Testing Bundle' target, and this interfered with existing SimplenoteScreenshots.

It's the first PR in my life :-)

### Release
These changes do not require release notes.
